### PR TITLE
fix(getMessages): Emit Error object in "error" event

### PR DESCRIPTION
**Pull request by frobinsonj**
Didn't really know how to title this, apologies if I got it wrong.

Error event `err` param should be type `Error` - https://github.com/abalabahaha/eris/blob/dev/index.d.ts#L711

### CODE REVIEW

**Comment by DonovanDMC** [comment]
It would probably be better to construct the error object, then overwrite the stack
Or even just override the message of the caught error then emit it rather than both making a new error object & emitting the original stack

File: lib/Client.js [hunk 1]
```diff
@@ -2673,7 +2673,7 @@ class Client extends EventEmitter {
             try {
                 return new Message(message, this);
             } catch(err) {
-                this.emit("error", `Error creating message from channel messages\n${err.stack}\n${JSON.stringify(messages)}`);
+                this.emit("error", new Error(`Error creating message from channel messages\n${err.stack}\n${JSON.stringify(messages)}`));
```
Comment (DonovanDMC) [inline lib/Client.js]:
```suggestion
                err.message = `Error creating message from channel messages\n${JSON.stringify(messages)}`;
                this.emit("error", err);
```

**Review by DonovanDMC — APPROVED** [review]

**Review by DonovanDMC — CHANGES_REQUESTED** [review]
[options.cause](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error) was added in node 16.9.0, we're currently on v10 (v14 at the latest), so we cannot use it

**Comment by frobinsonj** [comment]
> [options.cause](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error) was added in node 16.9.0, we're currently on v10 (v14 at the latest), so we cannot use it

Went with your original suggestion. Thanks.

**Review by DonovanDMC — APPROVED** [review]

**Review by abalabahaha — CHANGES_REQUESTED** [review]
Setting `.message` doesn't affect `.stack`. TBH I don't think the full JSON belongs in the error message. `warn` (also not great) and `debug` events are used to log JSON for other parsing errors

**Comment by DonovanDMC** [comment]
We can slice the first line off the stack and remake it with the new message

or we can just ignore the original stack and construct a new error

**Comment by bsian03** [comment]
Any other alternatives to this? Current solution (although error might be hard to read) is the best one imo

**Pull request closed without merging**